lib/pundit/authorization.rb

```diff
@@ -98,9 +98,14 @@ def policy_scope(scope, policy_scope_class: nil)
     #
     # @see https://github.com/varvet/pundit#policies
     # @param record [Object] the object we're retrieving the policy for
+    # @param policy_class [Class] the policy class we want to force use of
     # @return [Object, nil] instance of policy class with query methods
-    def policy(record)
-      policies[record] ||= Pundit.policy!(pundit_user, record)
+    def policy(record, policy_class: nil)
+      policies[{ policy_class: policy_class, record: record }] ||= if policy_class
```

Comment:
While `policies` is technically marked as private, I'm cautious against modifying its behaviour/content.

We probably should cache when `policy_class` is passed somehow, but that's a bigger consideration than adding the feature to `permitted_attributes`.